finetune/base_models/bert/featurizer.py

```diff
@@ -43,36 +57,80 @@ def bert_featurizer(X, encoder, config, train=False, reuse=None, **kwargs):
-        tf.expand_dims(tf.range(tf.cast(seq_length, tf.float32), dtype=tf.float32), 0),
-        axis=1
+        tf.cast(delimiters, tf.float32)
+        * tf.expand_dims(
+            tf.range(tf.cast(seq_length, tf.float32), dtype=tf.float32), 0
+        ),
+        axis=1,
     )
 
     mask = tf.sequence_mask(lengths, maxlen=seq_length, dtype=tf.float32)
     if config.num_layers_trained not in [config.n_layer, 0]:
-        raise ValueError("Bert base model does not support num_layers_trained not equal to 0 or n_layer")
+        raise ValueError(
+            "Bert base model does not support num_layers_trained not equal to 0 or n_layer"
+        )
 
-    with tf.variable_scope('model/featurizer', reuse=reuse):
+    with tf.variable_scope("model/featurizer", reuse=reuse):
         bert = BertModel(
             config=bert_config,
             is_training=train,
             input_ids=input_ids,
             input_mask=mask,
             token_type_ids=token_type_ids,
             use_one_hot_embeddings=False,
-            scope=None
+            scope=None,
+        )
+
+        embed_weights = bert.get_embedding_table()
+        features = tf.reshape(
+            bert.get_pooled_output(),
+            shape=tf.concat((initial_shape[:-2], [config.n_embed]), 0),
         )
+        sequence_features = tf.reshape(
+            bert.get_sequence_output(),
+            shape=tf.concat((initial_shape[:-1], [config.n_embed]), 0),
+        )
+
+        if config.use_auxiliary_info:
+            context_embed_weights = tf.get_variable(
+                name="ce",
+                shape=[context_dim, config.n_c_embed],
+                initializer=tf.random_normal_initializer(stddev=config.weight_stddev),
+            )
+
+            context_weighted_avg = tf.get_variable(
+                name="cwa",
+                shape=[context_dim],
+                initializer=tf.random_normal_initializer(stddev=config.weight_stddev),
+            )
+
+            if train:
+                context_embed_weights = dropout(
+                    context_embed_weights, config.embed_p_drop
+                )
+
+            with tf.variable_scope("context_embedding"):
+                weighted_C = tf.multiply(
+                    context, context_weighted_avg
+                )  # [batch_size, seq_length, context_dim] * [context_dim] = [batch_size, seq_length, context_dim], with weighted inputs
+                c_embed = tf.tensordot(
+                    weighted_C, context_embed_weights, axes=[[2], [0]]
+                )  # [batch_size, seq_length, context_dim] * [context_dim, n_embed] = [batch_size, seq_length, n_embed]
+                c_embed = layer_norm(c_embed, tf.get_variable_scope())
+                print(config.n_c_embed)
+                print(sequence_features)
+                sequence_features = tf.concat([sequence_features, c_embed], axis=2)
+                print(sequence_features)
+                c_embed = tf.reduce_sum(c_embed, axis=1)
```

Comment:
Hrmmm I don't want c_embed to have it's magnitude depend on sequence length.  Should probably be mean.